sklearn/decomposition/tests/test_fastica.py

```diff
@@ -220,10 +223,14 @@ def test_inverse_transform():
                 (False, n2): (n_features, n2)}
     for whiten in [True, False]:
         for n_components in [n1, n2]:
+            n_components_ = (n_components if n_components is not None else
+                             X.shape[1])
             ica = FastICA(n_components=n_components, random_state=rng,
                           whiten=whiten)
-            Xt = ica.fit_transform(X)
-            expected_shape = expected[(whiten, n_components)]
+            with warnings.catch_warnings(record=True):
```

Comment:
That would also ignore other possible warnings that we might want to know about.